### PR TITLE
Bug Fix : Updating colors makes the game state disappear

### PR DESCRIPTION
**Pull request by marsian83**
## The Issue
The colors changing option for lines, points and background caused the entire game state to disappear after the changing the color

## How to replicate the bug
1. Launch the activity
2. Marks a few lines and own a few square
3. Change any one of the colors
4. Notice that the entire board has disappeared 

## Demonstration
[click for demo](https://imgbox.com/mzsPQ7S6.gif)

The colors changing option for lines, points and background caused the entire game state to disappear after the changing the color This was due to the board being cleared and the newer lines having the new colors but the older lines not being redrawn

I have added a function called  `draw_board` on the `Game` class which re-renders all the lines and also the owners of each boxes (if they are owned)

calling the function every time any color is changed ensures that the state is visible even after changing the colors

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -122,15 +122,38 @@ def calc_grid_cant(self):
...
-
+        self.draw_board()
+
+    def draw_board(self):
+        for box_row in self.boxes:
+            for box in box_row:
+                box_tr = (box.pos_x + self.box_size[0]/2, box.pos_y - self.box_size[1]/2)
+                box_tl = (box.pos_x - self.box_size[0]/2, box.pos_y - self.box_size[1]/2)
+                box_br = (box.pos_x + self.box_size[0]/2, box.pos_y + self.box_size[1]/2)
+                box_bl = (box.pos_x - self.box_size[0]/2, box.pos_y + self.box_size[1]/2)
```
Comment (chimosky) [inline main.py]: Could you use floor division when dividing by ints?

**Comment by chimosky** [comment]
Reviewed, not tested.

File: main.py (hunk 1 above)
Comment (marsian83) [reply main.py]: yes, I have changed these to floor division as we're playing pixel coordinates

**Comment by quozl** [comment]
Thanks.  I think there is more to be done here.  You are adding a draw board method but there was already a draw grid method.  It may have been unfinished.  The new game button does not work properly; it clears the display, but previously selected bars between dots are unavailable.  Clearing the game on colour change is unfortunate, but reasonable.  If you're removing that behaviour, then really need to make sure new game button works properly!  Resize board also has relevance.  Some resizes could be allowed.  I've pushed some other commits, please cherry-pick them into your branch.

* 732ada8 ("Fix missing preview request"),
* d2bc37b ("Handle quit during end game"),
* 8c489cf ("Fix missing stop button response").

It looks like this activity was not finished.

**Comment by marsian83** [comment]
Hi @quozl due to my university exams I was not really able to look into these but now we're back on track and I have added the `preview request` commit into mine so that it can be merged 

Also, about the other changes, I plan on working on this activity further and I feel like to those changes would better fit when I have worked with the other parts of the code I plan to modify a little

As you mentioned the activity does look unfinished and I will be looking into fixing the restart system as well as the game quit section asap

**Comment by quozl** [comment]
[7969300](https://github.com/sugarlabs/dotsAndBoxes/pull/19/commits/79693003ff5fd511434ad5212084e610590bb8b2) seems to have lost myself as author.  Perhaps you didn't use "git cherry-pick".  Oh well.  Sure, let's see what you can do about the New Game feature.

**Comment by marsian83** [comment]
hi @quozl the new game button now works

**Comment by quozl** [comment]
Thanks.  [b99e9c8](https://github.com/sugarlabs/dotsAndBoxes/pull/19/commits/b99e9c8bb02b3b7c11445d6813728ea78771522f) raised some concerns with me;
* does not use 0 as the class box initialiser does,
* does not clear the owner,
* does not affect `vertical` or `horizontal` lists,
* different to how `set_board_size` clears the game,
* does not share code with `set_board_size`.

**Comment by marsian83** [comment]
Hi @quozl  I have refactored the code so that there is only one function reset_game which sets `horizontal`, `vertical` and `boxes` to empty lists as it was done in `set_board_size`
Then this is the function which is called in both `set_board_size` and `reset_game`

Clear game and Reset game are different as one only clears the game while the other resets the game which also includes redrawing the board. this is dont just to make the code more modular by introducing `clear_game` function which can be used for any other necessity if it may arise in the future.

**Comment by quozl** [comment]
Thanks.

**Pull request merged**